akka-persistence-query/src/main/scala/akka/persistence/query/journal/leveldb/EventsByPersistenceIdPublisher.scala

```diff
@@ -181,7 +181,7 @@ private[akka] class CurrentEventsByPersistenceIdPublisher(
     deliverBuf()
     if (highestSeqNr < toSequenceNr)
       toSeqNr = highestSeqNr
-    if (highestSeqNr == 0L || (buf.isEmpty && currSeqNo > toSequenceNr))
+    if (highestSeqNr == 0L || (buf.isEmpty && currSeqNo > toSequenceNr) || currSeqNo == fromSequenceNr)
```

Comment:
I've tried to remove `highestSeqNr == 0L` and use `buf.isEmpty && (currSeqNo > toSequenceNr || currSeqNo == fromSequenceNr)`.
Tests are working, I've added some more cases.